src/main/java/frc/robot/subsystems/Climber.java

```diff
@@ -0,0 +1,174 @@
+import com.nrg948.preferences.RobotPreferences;
+import com.nrg948.preferences.RobotPreferences.DoubleValue;
+import com.nrg948.preferences.RobotPreferencesLayout;
+import com.nrg948.preferences.RobotPreferencesValue;
+import edu.wpi.first.math.MathUtil;
+import edu.wpi.first.networktables.GenericEntry;
+import edu.wpi.first.util.datalog.BooleanLogEntry;
+import edu.wpi.first.util.datalog.DataLog;
+import edu.wpi.first.util.datalog.DoubleLogEntry;
+import edu.wpi.first.wpilibj.DataLogManager;
+import edu.wpi.first.wpilibj.DutyCycleEncoder;
+import edu.wpi.first.wpilibj.shuffleboard.BuiltInLayouts;
+import edu.wpi.first.wpilibj.shuffleboard.Shuffleboard;
+import edu.wpi.first.wpilibj.shuffleboard.ShuffleboardLayout;
+import edu.wpi.first.wpilibj.shuffleboard.ShuffleboardTab;
+import edu.wpi.first.wpilibj2.command.Commands;
+import edu.wpi.first.wpilibj2.command.SubsystemBase;
+import frc.robot.Constants.RobotConstants;
+import frc.robot.util.MotorController;
+import frc.robot.util.TalonFXAdapter;
+
+@RobotPreferencesLayout(groupName = "Climber", row = 0, column = 7, width = 1, height = 1)
+public class Climber extends SubsystemBase implements ShuffleboardProducer, ActiveSubsystem {
+  @RobotPreferencesValue
+  public static final RobotPreferences.BooleanValue ENABLE_TAB =
+      new RobotPreferences.BooleanValue("Climber", "Enable Tab", false);
+
+  private static final DataLog LOG = DataLogManager.getLog();
+
+  private final double MIN_ANGLE = -89;
+  private final double MAX_ANGLE = 150; // TODO: verify real max angle
+  private final double GEAR_RATIO = 5 * 5 * 66 / 18;
+  private final double ABSOLUTE_ENCODER_ZERO_OFFSET =
+      Math.toRadians(
+          360 - 173.1); // 360 - needed bcs abs encoder inversion is applied before offset
+
+  @RobotPreferencesValue
+  public static DoubleValue CLIMB_POWER = new DoubleValue("Climber", "Climb Power", 0.3);
+
+  @RobotPreferencesValue
+  public static DoubleValue NO_LOAD_POWER = new DoubleValue("Climber", "No Load Power", 0.3);
+
+  @RobotPreferencesValue
+  public static DoubleValue TOLERANCE_DEG = new DoubleValue("Climber", "Tolerance (deg)", 3);
+
+  @RobotPreferencesValue
+  public static DoubleValue CLIMB_GOAL_ANGLE_DEG =
+      new DoubleValue("Climber", "Climb Goal Angle (deg)", -88);
+
+  private double currentAngle;
+  private double goalAngle;
+  private boolean enabled;
+  private boolean isClimbing = false;
+
+  private final DutyCycleEncoder absoluteEncoder;
+
+  private MotorController mainMotor =
+      new TalonFXAdapter(
+          new TalonFX(RobotConstants.CAN.TalonFX.CLIMBER_MAIN_MOTOR_ID, "rio"),
+          false,
+          true,
+          1); // filler value; motor encoder value is not used.
+
+  @SuppressWarnings("unused")
+  private MotorController follower =
+      mainMotor.createFollower(RobotConstants.CAN.TalonFX.CLIMBER_FOLLOWER_MOTOR_ID, false);
+
+  private DoubleLogEntry logCurrentAbsoluteAngle =
+      new DoubleLogEntry(LOG, "Climber/Current Absolute Angle");
```

Comment:
Nit: everything being logged is assumed to be a "current" value.
```suggestion
      private DoubleLogEntry logAbsoluteAngle = new DoubleLogEntry(LOG, "Climber/Absolute Angle");
```